src/cli/java/org/commcare/util/screen/MultiSelectEntityScreen.java

```diff
@@ -63,6 +65,45 @@ public void updateSelection(String input, TreeReference[] selectedRefs)
         }
     }
 
+    @Override
+    protected boolean shouldAutoSelect() {
+        return mNeededDatum.isAutoSelectEnabled();
+    }
+
+    @Override
+    protected void autoSelectEntities() {
+        int selectionSize = references.size();
+        if (selectionSize == 0) {
+            throw new RuntimeException(getNoEntitiesError());
+        } else if (selectionSize > maxSelectValue) {
+            throw new RuntimeException(getMaxSelectError(selectionSize));
+        } else {
+            String[] evaluatedValues = new String[selectionSize];
+            for (int i = 0; i < selectionSize; i++) {
+                evaluatedValues[i] = getReturnValueFromSelection(references.elementAt(i));
+            }
+            processSelectionIntoInstance(evaluatedValues);
+        }
+    }
+
+    private String getMaxSelectError(int selectionSize) {
+        try {
+            return Localization.get("max.entity.selection.error",
```

Comment:
I think our naming convention for locale string is to name them according to location of origin so I think these should be something like:
```
case_list.auto_select.max_selection_error
case_list.auto_select.no_selection
```